docs/installation/est/Installing_EST_pkispawn.md

```diff
@@ -0,0 +1,227 @@
+est_realm_auth_type=BasicAuth
+est_realm_bind_dn=cn=Directory Manager
+est_realm_bind_password=
+est_realm_nickname=
+est_realm_user=
+est_realm_username=
+est_realm_password=
+est_realm_users_dn=ou=people,dc=est,dc=pki,dc=example,dc=com
+est_realm_groups_dn=ou=groups,dc=est,dc=pki,dc=example,dc=com
+est_realm_statements=/usr/share/pki/est/conf/realm/postgresql/statements.conf
+est_authorizer_exec_path=/usr/share/pki/est/bin/estauthz
+```
+
+The `est_ca_*` provides information related to the user and profile
+configured in the CA for the EST subsystem.
+
+The `est_authorizer_exec_path` is the executable responsible to verify
+the authorisation. The provided script checks only that the user has
+the role *estclient*.
+
+The `est_realm_*` options allow to customise the realm. Possible types
+are: ds, postgresql and in-memory. As an example, to install EST with
+PostgreSQL the command will be
+
+```
+# pkispawn \
+    -f /usr/share/pki/server/examples/installation/est.cfg \
+    -s EST \
+    -D est_realm_url="jdbc:postgresql://postgresql.example.com:5432/est?ssl=true&sslmode=require" \
+    -D est_realm_type=postgresql \
+    -D est_realm_user=est \
+    -D est_realm_password=mysecretpassword \
+    -v
+```
+
+The `est_realm_custom` is a path to a custom realm configuration for
+tomcat and if provided it will overwrite all other realm related
+configurations.
+
+### Installation on separate instance
+
+When installed on a separate instance the CA URL and the signing
+certificate has to be provided. This is done with the command:
+
+```
+# pkispawn \
+    -f /usr/share/pki/server/examples/installation/est.cfg \
+    -s EST \
+    -D est_realm_url=ldap://estds.example.com:3389 \
+    -D pki_ca_uri=https://ca.example.com:8443 \
+    -D pki_cert_chain_path=ca_signing.crt \
+    -D pki_cert_chain_nickname=caSigning \
+    -v
+```
+
+Note: the CA signing certificate can be obtained with the following command in the CA server:
+```
+# pki-server cert-export ca_signing --cert-file ca_signing.crt
+```
+
+By default, the EST server cert will be issued based on the
+information provided via the `pkispawn` parameters `est_ca_profile`
+and `est_ca_user_*`.  Alternatively, the EST server cert (and the user
+certificate if it is needed to connect to the CA) could be pre-issued
+prior to `pkispawn` with its full chain in a **PKCS#12** bundle supplied
+via the `pki_server_pkcs12_*` parameters on the `pkispawn` command
+line as exemplified below. It is important that the certificate
+aliases in the PKCS#12 match with the nickname used by EST. For SSL
+certificate the nickname configured in `est.cfg` is `sslserver` but
+can be modified. The command to deploy with the certificate bundle is:
+
+```
+# pkispawn \
+    -f /usr/share/pki/server/examples/installation/est.cfg \
+    -s EST \
+    -D est_realm_url=ldap://estds.example.com:3389 \
+    -D pki_ca_uri=https://ca.example.com:8443 \
+    -D est_ca_user_password= \
+    -D est_ca_user_certificate=estUser \
```

Comment:
@fmarco76 looks like you updated while I was typing up the following:   Just realized that I missed the "(and the user certificate if it is needed to connect to the CA)" part in my suggestion clarification.  So the est_ca_user_* are for client cert talking to the CA then? Which profile is it using (which parameter in pkispawn cfg?)?  As for it's alternative, I see in the example below, est_ca_user_password and est_ca_user_certificate are supplied, but it doesn't seem to point to any p12.  How does it work?

Reply:
> So the est_ca_user_* are for client cert talking to the CA then? 
These parameter are to configure the CA user for EST communication. This is created in the prerequisite described in the main installation document and added here. 
 
> Which profile is it using (which parameter in pkispawn cfg?)

The profile is defined with the option `est_ca_profile` (line 107) and default it is _estServiceCert_.

>As for it's alternative, I see in the example below, est_ca_user_password and est_ca_user_certificate are supplied, but it doesn't seem to point to any p12. How does it work?

`est_ca_user_certificate` is the name of the certificate to use inside the nssdb of the instance. `est_ca_user_password` is empty to be sure it is not used but if both are provided I think the certificate has precedence but it is in the EST code so I should verify.

Reply:
@fmarco76 sorry I'm still a bit lost.  Sounds like you are saying that the EST server cert and the EST client cert (the one to be associated with the est user in the CA agent group) are using the same enrollment profile (I hope not), est**Server**Cert?  Oh wait, you said est**Service**Cert.  So there are two profiles, one being estServerCert (the one for server) and the other being estServiceCert (the one for client)?  I could see how the EST server cert is being issued, either via the default set in est_ca_profile=, or by pkispawn command line -D pki_server_pkcs12_*, but I still don't get how the EST client cert (for talking to the ca) got issued.  Unless I missed it, I do not see it being instructed to be issued in the prerequisite section.  Could you please point to me where that is?